rs/index/src/ivf/index.rs

```diff
@@ -13,7 +15,7 @@ use crate::posting_list::combined_file::FixedIndexFile;
 use crate::utils::{IdWithScore, SearchContext};
 use crate::vector::fixed_file::FixedFileVectorStorage;
 
-pub struct Ivf<Q: Quantizer> {
+pub struct Ivf<Q: Quantizer, D: IntSeqDecoderIterator> {
```

Comment:
a better way to do this is to use have a decoder factory. so you'll have something like:
```
pub struct<E: Encoding> IteratorFactory {
    _phantom: PhantomData<E>,
}

impl IteratorFactor {
    pub fn get_iterator() -> Iterator {
        ....
    }
}
```

Reply:
you can probably use [associated type](https://doc.rust-lang.org/rust-by-example/generics/assoc_items/types.html) to restrict what type of iterator will be produced based on what encoding